hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -1680,6 +1680,25 @@ case class TableToTableApply(child: TableIR, function: TableToTableFunction) ext
   }
 }
 
+case class BlockMatrixToTableApply(
+  bm: BlockMatrixIR,
+  aux: IR,
+  function: BlockMatrixToTableFunction) extends TableIR {
+
+  override lazy val children: IndexedSeq[BaseIR] = Array(bm, aux)
+
+  override def copy(newChildren: IndexedSeq[BaseIR]): TableIR =
+    BlockMatrixToTableApply(
+      newChildren(0).asInstanceOf[BlockMatrixIR],
+      newChildren(1).asInstanceOf[IR],
+      function)
+
+  override lazy val typ: TableType = function.typ(bm.typ, aux.typ)
+
+  protected[ir] override def execute(ctx: ExecuteContext): TableValue =
+    function.execute(ctx, bm.execute(ctx), Interpret[Any](ctx, aux))
```

Comment:
I think Interpret is the wrong thing to use here. I think it should be 

```
CompileAndEvaluate(ctx, aux, optimize = false)
```

We really need to clean up the execution control flow.